lightning/src/routing/router.rs

```diff
@@ -333,79 +429,199 @@ pub fn get_route<L: Deref>(our_node_id: &PublicKey, network: &NetworkGraph, targ
+				let mut res = vec!(new_entry.route_hop.clone());
+				let mut path_bottleneck_msat = None;
+				let mut cur_path_fees = Vec::new();
+				let mut smallest_channel_id = None;
+
+				loop {
+					if let Some(&(_, ref features)) = first_hop_targets.get(&res.last().unwrap().pubkey) {
+						res.last_mut().unwrap().node_features = features.to_context();
+					} else if let Some(node) = network.get_nodes().get(&res.last().unwrap().pubkey) {
+						if let Some(node_info) = node.announcement_info.as_ref() {
+							res.last_mut().unwrap().node_features = node_info.features.clone();
+						} else {
+							res.last_mut().unwrap().node_features = NodeFeatures::empty();
+						}
 					} else {
-						res.last_mut().unwrap().node_features = NodeFeatures::empty();
+						// We should be able to fill in features for everything except the last
+						// hop, if the last hop was provided via a BOLT 11 invoice (though we
+						// should be able to extend it further as BOLT 11 does have feature
+						// flags for the last hop node itself).
+						assert!(res.last().unwrap().pubkey == *target);
 					}
-				} else {
-					// We should be able to fill in features for everything except the last
-					// hop, if the last hop was provided via a BOLT 11 invoice (though we
-					// should be able to extend it further as BOLT 11 does have feature
-					// flags for the last hop node itself).
-					assert!(res.last().unwrap().pubkey == *target);
+
+					// For this path, select the channel with the lowest available amount to avoid hitting it
+					// in the next path finding iterations.
+					if let Some(current_hop_bottleneck_msat) = new_entry.available_msat {
+						if let Some(full_path_bottleneck_msat) = path_bottleneck_msat {
+							if current_hop_bottleneck_msat < full_path_bottleneck_msat {
+								path_bottleneck_msat = Some(current_hop_bottleneck_msat);
+								smallest_channel_id = Some(res.last().unwrap().short_channel_id);
+							}
+						}
+					}
+
+					if res.last().unwrap().pubkey == *target {
+						break;
+					}
+
+					new_entry = match dist.remove(&res.last().unwrap().pubkey) {
+						Some(hop) => hop,
+						None => return Err(LightningError{err: "Failed to find a non-fee-overflowing path to the given destination".to_owned(), action: ErrorAction::IgnoreError}),
+					};
+					res.last_mut().unwrap().fee_msat = new_entry.route_hop.fee_msat;
+					res.last_mut().unwrap().cltv_expiry_delta = new_entry.route_hop.cltv_expiry_delta;
+					res.push(new_entry.route_hop);
+					cur_path_fees.push(new_entry.channel_fees);
 				}
-				if res.last().unwrap().pubkey == *target {
-					break;
+				res.last_mut().unwrap().fee_msat = final_value_msat;
+				res.last_mut().unwrap().cltv_expiry_delta = final_cltv;
+
+				// Since the paths allows to transfer as much sats as the lowest channel it has ("bottleneck"),
+				// we should recompute the fees so that we don't overpay with the other (larger) used channels.
```

Comment:
What the relation between a lowest channel and overpaying fees with other used channels among the same path ? Or do you mean comparing paths between themselves ? (intra or inter-path comparisons ?)

Reply:
I don't understand what you missing here. It's intra-path. Recompute fees along a given path considering that at every channel only the path-bottleneck amount will be used (and not the full amount of that channel).

I don't see why inter-path should matter in this case.

Reply:
Ah I see but the relay-feerate shouldn't change based on the amount we send through. Only the absolute fee.

"paths" should be singular of channels should be plural ?